ray-operator/apis/ray/v1alpha1/rayservice_types.go

```diff
@@ -1,22 +1,85 @@
+
+const (
+	FAIL_TO_GET_RAYSERVICE           ServiceStatus = "fail_to_get_rayservice"
+	FAIL_TO_GET_OR_CREATE_RAYCLUSTER ServiceStatus = "fail_to_get_or_create_RAYCLUSTER"
+	WAIT_FOR_DASHBOARD               ServiceStatus = "wait_for_dashboard"
+	FAIL_SERVE_DEPLOY                ServiceStatus = "fail_serve_deploy"
+	FAIL_GET_SERVE_DEPLOYMENT_STATUS ServiceStatus = "fail_get_serve_deployment_status"
+	RUNNING                          ServiceStatus = "running"
+	RESTARTING                       ServiceStatus = "restarting"
+	FAIL_DELETE_RAYCLUSTER           ServiceStatus = "fail_delete_raycluster"
+)
+
+// ServeConfigSpec defines the desired state of RayService
+type ServeConfigSpec struct {
+	Name                      string             `json:"name"`
+	ImportPath                string             `json:"import_path"`
+	InitArgs                  []string           `json:"init_args,omitempty"`
+	InitKwargs                map[string]string  `json:"init_kwargs,omitempty"`
+	NumReplicas               *int32             `json:"num_replicas,omitempty"`
+	RoutePrefix               string             `json:"route_prefix,omitempty"`
+	MaxConcurrentQueries      *int32             `json:"max_concurrent_queries,omitempty"`
+	UserConfig                map[string]string  `json:"user_config,omitempty"`
+	AutoscalingConfig         map[string]string  `json:"autoscaling_config,omitempty"`
+	GracefulShutdownWaitLoopS *float64           `json:"graceful_shutdown_wait_loop_s,omitempty"`
+	GracefulShutdownTimeoutS  *float64           `json:"graceful_shutdown_timeout_s,omitempty"`
+	HealthCheckPeriodS        *float64           `json:"health_check_period_s,omitempty"`
+	HealthCheckTimeoutS       *float64           `json:"health_check_timeout_s,omitempty"`
+	RayActorOptions           RayActorOptionSpec `json:"ray_actor_options,omitempty"`
+}
+
+// RayActorOptionSpec defines the desired state of RayActor
+type RayActorOptionSpec struct {
+	RuntimeEnv        map[string][]string `json:"runtime_env,omitempty"`
+	NumCpus           *float64            `json:"num_cpus,omitempty"`
+	NumGpus           *float64            `json:"num_gpus,omitempty"`
+	Memory            *int32              `json:"memory,omitempty"`
+	ObjectStoreMemory *int32              `json:"object_store_memory,omitempty"`
+	Resources         map[string]string   `json:"resources,omitempty"`
+	AcceleratorType   string              `json:"accelerator_type,omitempty"`
+}
+
 // RayServiceSpec defines the desired state of RayService
 type RayServiceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	HealthCheckProbe *v1.Probe         `json:"healthCheckConfig,omitempty"`
+	ServeConfigSpecs []ServeConfigSpec `json:"serveConfigs,omitempty"`
+	RayClusterSpec   RayClusterSpec    `json:"rayClusterConfig,omitempty"`
+}
+
+// ServeStatus defines the desired state of Serve Deployment
+type ServeStatus struct {
```

Comment:
Could we rename this to `ServeDeploymentStatus` or `DeploymentStatus`? We're introducing a new [Serve Application-level status](https://github.com/ray-project/ray/blob/16bdfe6a3981a08a02a66984dadeac5b73c7cb1f/python/ray/serve/schema.py#L358-L365) in the future, so this name might clash with that.

Reply:
sgtm